instrumentation/instaamqp/propagation.go

```diff
@@ -0,0 +1,78 @@
+// (c) Copyright IBM Corp. 2022
```

Comment:
Just as an idea:

I think we can avoid having two implementations of the carriers and I see two options:

1. I believe we are interested not in particular types we wrap, but underlying headers -> so we can think about single carrier implementation with `Headers amqp.Table{}` member.

2. Second option is to use `ot.TextMapCarrier`, but before that we need to convert `amqp.Table{}` to TextMap (map[string]interface to map[string]string)

What do you think?

Reply:
I will mark this suggestion as resolved as we have agreed on merging the carriers into one and applying the necessary changes.
We may start a new thread for the implementation itself.